src/routes/jira/connect/enterprise/jira-connect-enterprise-post.ts

```diff
@@ -1,64 +1,71 @@
 import { Request, Response } from "express";
 import { GitHubServerApp } from "models/github-server-app";
-import { isValidUrl } from "utils/is-valid-url";
+import { validateUrl } from "utils/validate-url";
 import { statsd } from "config/statsd";
 import { metricError } from "config/metric-names";
 import { sendAnalytics } from "utils/analytics-client";
 import { AnalyticsEventTypes, AnalyticsTrackEventsEnum } from "interfaces/common";
 import { createAnonymousClient } from "utils/get-github-client-config";
 
-const TIMEOUT_PERIOD_MS = 30 * 1000;
+const GITHUB_CLOUD_HOSTS = ["github.com", "www.github.com"];
 
-interface MessageAndCode {
-	errorCode: string;
-	message: string;
+enum ErrorResponseCode {
+	InvalidUrl = "GHE_ERROR_INVALID_URL",
+	CloudHost = "GHE_ERROR_GITHUB_CLOUD_HOST",
+	CannotConnect = "GHE_ERROR_CANNOT_CONNECT"
 }
 
-interface GheServerUrlErrors {
-	[key: string | number]: MessageAndCode;
+function isInteger(n: string) {
+	return !isNaN(Number(n));
 }
 
-const GHE_ERROR_UNREACHABLE = {
-	errorCode: "GHE_ERROR_ENOTFOUND",
-	message: "Request to URL failed"
-};
-
-export const gheServerUrlErrors: GheServerUrlErrors = {
-	invalidUrl: {
-		errorCode: "GHE_ERROR_INVALID_URL",
-		message: "Invalid URL"
-	},
-	ENOTFOUND: GHE_ERROR_UNREACHABLE,
-	DEPTH_ZERO_SELF_SIGNED_CERT: GHE_ERROR_UNREACHABLE,
-	403: GHE_ERROR_UNREACHABLE,
-	502: {
-		errorCode: "GHE_SERVER_BAD_GATEWAY",
-		message: "Bad gateway"
-	},
-	ECONNABORTED: {
-		errorCode: "GHE_ERROR_CONNECTION_TIMED_OUT",
-		message: "Connection timed out"
-	}
-};
-
 export const JiraConnectEnterprisePost = async (
 	req: Request,
 	res: Response
 ): Promise<void> => {
+
+	// Must be configurable and re-evaluated on each execution for testing, therefore
+	// inside the handler
+	const TIMEOUT_PERIOD_MS = parseInt(process.env.JIRA_CONNECT_ENTERPRISE_POST_TIMEOUT_MSEC || "30000");
```

Comment:
That's just a hackery for the test. I don't want to add it to envVars because it shouldn't be modified elsewhere... Open to other ideas how to speed up the test and not to wait 30 seconds ;)